### PR TITLE
Re-enable most of the navigation block e2e tests

### PR DESCRIPTION
**Pull request by talldan**
## Description
This re-enables most of the navigation block's end to end tests.

The tests had to be re-written when the navigation switched to storing its content in a post.

The tests now use less response mocking and more use of the REST end to end test utility for creating and destroying down data.

I've also refactored some other aspects of the tests.

I'll do a part 2 of this to look at enabling the remaining tests.

### CODE REVIEW

File: packages/e2e-test-utils/src/ensure-sidebar-opened.js [hunk 1]
```diff
@@ -4,14 +4,11 @@
...
-	} catch ( error ) {
-		return page.click(
-			'.edit-post-header__settings [aria-label="Settings"]'
-		);
+	const toggleSidebarButton = await page.$(
+		'.edit-post-header__settings [aria-label="Settings"][aria-expanded="false"]'
+	);
+
+	if ( toggleSidebarButton ) {
+		await toggleSidebarButton.click();
```
Comment (talldan) [inline packages/e2e-test-utils/src/ensure-sidebar-opened.js:12]: The previous implementation wasn't working for some reason. This is a better way to do it anyway.

File: packages/e2e-test-utils/src/open-publish-panel.js [hunk 2]
```diff
@@ -2,9 +2,20 @@
  * Opens the publish panel.
  */
 export async function openPublishPanel() {
```
Comment (talldan) [inline packages/e2e-test-utils/src/open-publish-panel.js:4]: `openPublishPanel` and `publishPost` were never updated to handle the multi-entity saving flow, so I had to do that.

**Review by adamziel — APPROVED** [review]

File: packages/e2e-tests/specs/editor/blocks/navigation.test.js [hunk 3]
```diff
@@ -785,7 +676,7 @@ describe.skip( 'Navigation', () => {
 		expect( isScriptLoaded ).toBe( true );
 	} );
 
-	describe( 'Creating and restarting', () => {
+	describe.skip( 'Creating and restarting', () => {
```
Comment (adamziel) [inline packages/e2e-tests/specs/editor/blocks/navigation.test.js:679]: did it break?

Comment (talldan) [reply packages/e2e-tests/specs/editor/blocks/navigation.test.js:679]: I don't think so, but they also weren't enabled in trunk.

This branch predated these tests and I'd already removed most of the mocking utilities, so when I rebased I realised I had a lot more work to do 😄 

I'll take a look at switching them away from mocks and over to using the REST API in a fresh PR.

Thanks for the approval 😄

**Pull request merged**